src/http.c

```diff
@@ -279,7 +279,17 @@ int mg_http_parse(const char *s, size_t len, struct mg_http_message *hm) {
   hm->uri.buf = (char *) s;
   while (s < end && (n = clen(s, end)) > 0) s += n, hm->uri.len += n;
   while (s < end && s[0] == ' ') s++;  // Skip spaces
+  is_response = mg_ncasecmp(hm->method.buf, "HTTP/", 5) == 0;
```

Comment:
At this point the request can be invalid, and < 5 bytes. We must check for the size too to prevent invalid access:

```c
is_response = hm->method.pen > 5 && mg_ncasecmp(hm->method.buf, "HTTP/", 5) == 0;
```